### PR TITLE
add support for `start_dir` easyconfig template

### PR DESCRIPTION
**Pull request by jfgrimm**
(created using `eb --new-pr`)

### CODE REVIEW

**Comment by jfgrimm** [comment]
@ocaisa I'll take a look
trying it out atm, and it `%(srcdir)s` seems to always be `None`, even in `CMakeMake` easyconfigs

**Comment by ocaisa** [comment]
Actually yeah, that makes sense, for each source you would have a diffent source directory so there's no unique property that you can reference there (https://github.com/easybuilders/easybuild-framework/blob/develop/easybuild/framework/easyblock.py#L2434)

File: easybuild/framework/easyconfig/templates.py [hunk 1]
```diff
@@ -73,6 +73,8 @@
 TEMPLATE_NAMES_EASYBLOCK_RUN_STEP = [
     ('builddir', "Build directory"),
     ('installdir', "Installation directory"),
+    ('srcdir', "Source directory"),
+    ('start_dir', "Directory to start make in"),
```
Comment (ocaisa) [inline easybuild/framework/easyconfig/templates.py:77]:
```suggestion
    ('start_dir', "Directory in which the build process begins"),
```

**Comment by ocaisa** [comment]
It looks like `start_dir` is not going to work either since that isn't typically known until the `prepare_step` (which calls https://github.com/easybuilders/easybuild-framework/blob/develop/easybuild/framework/easyblock.py#L1976)

**Comment by boegel** [comment]
It's OK if `%(start_dir)s` can only be resolved after some step (like `prepare`), since that's also the case of other templates like `%(pyver)s`

**Comment by boegel** [comment]
superseded by #4073, so closing...

**Pull request closed without merging**